### PR TITLE
Implements frontend for deployment rollback

### PR DESCRIPTION
**Pull request by shu-mutou**
History list is added into deployment detail view, and `Rollback` action is added in each row except for current revision.

Fixes: #2889

### CODE REVIEW

**Comment by shu-mutou** [comment]
/hold
This would conflict with #6005. Wait for merging it.

File: src/app/backend/resource/deployment/rollout.go [hunk 1]
```diff
@@ -39,17 +45,45 @@ type RolloutSpec struct {
...
+
+	// Unordered list of Replica Sets.
+	HistoryList []History `json:"history"`
+
+	// List of non-critical errors, that occurred during resource retrieval.
+	Errors []error `json:"errors"`
+}
+
+// History is a presentation layer view of Kubernetes Replica Set resource for history.
+type History struct {
```
Comment (floreks) [inline src/app/backend/resource/deployment/rollout.go:63]: This kind of breaks our pattern of resource being a Kubernetes resource. It is then used in the frontend as a resource also. I'd have to rethink how to incorporate that into our patterns.

Comment (shu-mutou) [reply src/app/backend/resource/deployment/rollout.go:63]: Yes, these are defined in frontend also.
To create dialog contents and submission endpoint URL, the action on history row needed to pass deployment name and namespace to dialog. So I defined new type History that have ReplicaSet and parent resource information. Could you let me know if you have better idea about this?

Comment (floreks) [reply src/app/backend/resource/deployment/rollout.go:63]: I'll think about how we could add this without breaking our patterns. Resource should be mappable to a real Kubernetes resource. History/HistoryList is a very abstract thing.

**Comment by floreks** [comment]
I was on vacation last week. I'll try to take a look at it this week.

**Comment by cell4rom** [comment]
Hello, why You don't using Redux, it's very important in a project like kubernetes UI.

**Comment by floreks** [comment]
It's not important. It's a matter of personal preference. Also, PR is not a good place to discuss such things.

**Comment by shu-mutou** [comment]
@maciaszczykm @floreks PTAL again. Thanks!!

**Comment by floreks** [comment]
Since https://github.com/kubernetes/dashboard/pull/7047 has been merged and it introduces major architecture changes we should close this PR. If you are willing to work on it, you can open a new PR against the current master.

/close

**Pull request closed without merging**